geo/src/algorithm/triangulate_delaunay.rs

```diff
@@ -0,0 +1,645 @@
+            if super_tri_vertices.contains(pt) {
+                add_to_update = false;
+            }
+        }
+        if add_to_update {
+            new_triangles.push(tri.clone())
+        }
+    }
+    new_triangles
+}
+
+/// A triangle structure used during Delaunay Triangulation.
+#[derive(Debug, Clone, PartialEq)]
+pub struct DelaunayTriangle<T: GeoFloat>(Triangle<T>);
+
+impl<T: GeoFloat> From<Triangle<T>> for DelaunayTriangle<T> {
+    fn from(value: Triangle<T>) -> Self {
+        DelaunayTriangle(value)
+    }
+}
+
+impl<T: GeoFloat> From<DelaunayTriangle<T>> for Triangle<T> {
+    fn from(value: DelaunayTriangle<T>) -> Self {
+        value.0
+    }
+}
+
+fn is_line_shared<T: GeoFloat>(a: &Line<T>, b: &Line<T>) -> bool {
+    (a.start == b.start && a.end == b.end) || (a.start == b.end && a.end == b.start)
+}
+
+/// Methods required for delaunay triangulation
+impl<T: GeoFloat> DelaunayTriangle<T>
+where
+    f64: From<T>,
+{
+    #[cfg(feature = "voronoi")]
+    /// Check if a `DelaunayTriangle` is a neighbour i.e.
+    /// shares an edge.
+    /// If the triangles are neighbours the shared edge is returned.
+    pub(crate) fn shares_edge(&self, other: &DelaunayTriangle<T>) -> Option<Line<T>> {
+        let other_lines = other.0.to_lines();
+        for line in self.0.to_lines().iter() {
+            for other_line in other_lines.iter() {
+                if is_line_shared(line, other_line) {
+                    return Some(*line);
+                }
+            }
+        }
+        None
+    }
+
+    /// Check if a `Coord` is in the [Circumcircle](https://en.wikipedia.org/wiki/Circumcircle)
+    /// for the Delaunay triangle.
+    /// This method uses the determinant of the vertices of the triangle and the
+    /// new point as described by [Guibas & Stolfi](https://doi.org/10.1145%2F282918.282923)
+    /// and on [Wikipedia](https://en.wikipedia.org/wiki/Delaunay_triangulation#Algorithms).
+    pub(crate) fn is_in_circumcircle(&self, c: &Coord<T>) -> bool {
+        let a_d_x: f64 = (self.0 .0.x - c.x).into();
+        let a_d_y: f64 = (self.0 .0.y - c.y).into();
+        let b_d_x: f64 = (self.0 .1.x - c.x).into();
+        let b_d_y: f64 = (self.0 .1.y - c.y).into();
+        let c_d_x: f64 = (self.0 .2.x - c.x).into();
+        let c_d_y: f64 = (self.0 .2.y - c.y).into();
+        let a_d_x_d_y = a_d_x.powi(2) + a_d_y.powi(2);
+        let b_d_x_d_y = b_d_x.powi(2) + b_d_y.powi(2);
+        let c_d_x_d_y = c_d_x.powi(2) + c_d_y.powi(2);
+
+        // Compute the determinant of the following matrix
+        // [
+        //     [a_d_x, a_d_y, a_d_x_d_y],
+        //     [b_d_x, b_d_y, b_d_x_d_y],
+        //     [c_d_x, c_d_y, c_d_x_d_y],
+        // ]
+        //
+        let determinant = a_d_x * ((b_d_y * c_d_x_d_y) - (b_d_x_d_y * c_d_y))
+            - a_d_y * ((b_d_x * c_d_x_d_y) - (b_d_x_d_y * c_d_x))
+            + a_d_x_d_y * (b_d_x * c_d_y - b_d_y * c_d_x);
+
+        determinant > 0.0
```

Comment:
Why is the `f64` necessary? If it's just because of the comparison with `0.0` at the end, you can use `T::zero()` instead of that literal value

Reply:
it is not required.  I've removed.